src/Select.purs

```diff
@@ -288,18 +289,15 @@ handleAction handleExtraQuery handleMessage = case _ of
       raise' $ VisibilityChanged v
 
   AndThen act1 act2 -> do 
-    handleAction' act1 
-    handleAction' act2 
+    handleAction handleExtraQuery handleMessage act1 
+    handleAction handleExtraQuery handleMessage act2 
     pure unit
   
   RunQuery query -> do
     _ <- handleQuery handleExtraQuery query
     pure unit
   
   where
-  -- recursively evaluate an action
-  handleAction' = handleAction handleExtraQuery handleMessage
```

Comment:
This is interesting: defining this helper function causes a "too much recursion" error when the code is run. But not defining this helper and instead writing it out every time fixes the issue. I'd like to try to find a smaller reproduction as it's an irritating restriction.

Reply:
Ah -- no, as the next commit shows, you can also eta-expand it to get around the issue. Still annoying.